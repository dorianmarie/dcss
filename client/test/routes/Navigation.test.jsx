import React from 'react';
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useLayoutEffect: jest.requireActual('react').useEffect
}));

import {
  fetchImplementation,
  mounter,
  reduxer,
  serialize,
  snapshotter,
  state
} from '../bootstrap';
import { unmountComponentAtNode } from 'react-dom';

import {
  act,
  fireEvent,
  prettyDOM,
  render,
  screen,
  waitFor
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

/** @GENERATED: BEGIN **/

import Navigation from '../../routes/Navigation.jsx';
/** @GENERATED: END **/

const original = JSON.parse(JSON.stringify(state));
let container = null;
let commonProps = null;
let commonState = null;

beforeAll(() => {
  (window || global).fetch = jest.fn();
});

afterAll(() => {
  jest.restoreAllMocks();
});

beforeEach(() => {
  jest.useFakeTimers();
  container = document.createElement('div');
  container.setAttribute('id', 'root');
  document.body.appendChild(container);

  fetchImplementation(fetch);

  /** @GENERATED: BEGIN **/

  /** @GENERATED: END **/

  commonProps = {};
  commonState = JSON.parse(JSON.stringify(original));
});

afterEach(() => {
  jest.runOnlyPendingTimers();
  jest.useRealTimers();
  jest.resetAllMocks();
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  commonProps = null;
  commonState = null;
});

test('Navigation', () => {
  expect(Navigation).toBeDefined();
});

test('Render 1 1', async done => {
  /** @GENERATED: BEGIN **/
  const Component = Navigation;
  const props = {
    ...commonProps
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);
  /** @GENERATED: END **/

  const { asFragment } = render(<ConnectedRoutedComponent {...props} />);
  expect(asFragment()).toMatchSnapshot();

  done();
});
