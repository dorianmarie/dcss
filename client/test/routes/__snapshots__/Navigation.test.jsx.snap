// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render 1 1 1`] = `
<DocumentFragment>
  <div
    aria-controls="nav"
    class="ui borderless menu"
    id="nav"
  >
    <div
      class="item navigation__menu-item-logo"
      tabindex="0"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
    </div>
    <a
      aria-label="See all scenarios"
      class="item"
      href="/scenarios/"
      role="option"
      tabindex="0"
    >
      Scenarios
    </a>
    <a
      class="item"
      href="/cohorts"
      tabindex="0"
    >
      Cohorts
    </a>
    <a
      class="item"
      href="/history"
      tabindex="0"
    >
      History
    </a>
    <a
      class="item"
      href="/downloads"
      tabindex="0"
    >
      Downloads
    </a>
    <a
      class="item"
      href="/admin"
      tabindex="0"
    >
      Admin
    </a>
    <div
      class="right menu"
    >
      <div
        aria-expanded="false"
        class="ui item simple dropdown"
        role="listbox"
        tabindex="0"
      >
        <span
          class="ui medium text"
        />
         
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            class="item"
            role="option"
            tabindex="0"
          >
            Settings
          </div>
          <div
            class="item"
            role="option"
            tabindex="0"
          >
            Log out
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
